vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/network/externalipranger/externalip_admission_test.go

```diff
@@ -1,15 +1,45 @@
 package externalipranger
 
 import (
+	"fmt"
 	"net"
 	"strings"
 	"testing"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+
+	"k8s.io/apiserver/pkg/authentication/user"
+
 	kapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/serviceaccount"
 )
 
+type fakeTestAuthorizer struct {
+	t *testing.T
+}
+
+func fakeAuthorizer(t *testing.T) authorizer.Authorizer {
+	return &fakeTestAuthorizer{
+		t: t,
+	}
+}
+
+func (a *fakeTestAuthorizer) Authorize(attributes authorizer.Attributes) (authorizer.Decision, string, error) {
+	ui := attributes.GetUser()
+	if ui == nil {
+		return authorizer.DecisionNoOpinion, "", fmt.Errorf("No valid UserInfo for Context")
+	}
+	// system:serviceaccount:herpy:derpy user aka admin user is allowed to set
+	// external IPs
+	if ui.GetName() == "system:serviceaccount:herpy:derpy" {
```

Comment:
use a more self-documenting name. No one is going to understand why a test is expected to pass for "herpy:derpy" but fail for "slurpy:derpy".

eg `"system:serviceaccount:test:admin"` vs `"system:serviceaccount:test:ordinary-user"`

Reply:
I was following the example in the `podnodeconstraints` admission plugin, but I suppose I can change the account name for clarity. The fake authorizer comments do make it clear that `herpy:derpy` is an admin user, but i guess i can make it unambiguous further using appropriate account names. Thanks for keeping me honest.